evthread-internal.h

```diff
@@ -116,12 +116,16 @@ extern int evthread_lock_debugging_enabled_;
 /** Lock an event_base, if it is set up for locking.  Acquires the lock
     in the base structure whose field is named 'lockvar'. */
 #define EVBASE_ACQUIRE_LOCK(base, lockvar) do {				\
-		EVLOCK_LOCK((base)->lockvar, 0);			\
+		if (base) {										\
```

Comment:
From which places EVBASE_ACQUIRE_LOCK/EVBASE_RELEASE_LOCK can be called w/o base? Actually using global base if it is not passed is old and deprecated stuff, it should go away, so does the following checks.

Reply:
Hi, @azat,
1.  in function `event_base_once` ,  EVBASE_ACQUIRE_LOCK/EVBASE_RELEASE_LOCK can be called w/o base, and there's no guarantee for a check . 
2. Considering the second advice you propose, i also think it is better to return a failure if unexpected null base which means a FAIL but not use current base.  
And i will update the patch later.
thx.

Reply:
Hi, @azat, i have update the patch, add a guard before call EVBASE_ACQUIRE_LOCK/EVBASE_RELEASE_LOCK, but no perform in the macros. RFC.